gremlin-dotnet/src/Gremlin.Net/Process/Utils.cs

```diff
@@ -63,5 +66,23 @@ public static void Forget(this Task task)
                 t.Exception?.Handle(_ => true);
             }, TaskContinuationOptions.ExecuteSynchronously);
         }
+
+        /// <summary>
+        ///  Returns a user agent for connection request headers.
+        ///
+        /// Format:
+        /// "[Application Name] [GLV Name].[Version] [Language Runtime Version] [OS].[Version] [CPU Architecture]"
+        /// </summary>
+        private static string GenerateUserAgent()
+        {
+            var applicationName = Assembly.GetExecutingAssembly().GetName().Name.Replace(' ', '_');
```

Comment:
`Name` could be `null` which would throw a `NullReferenceException`. Please use conditional access:
```suggestion
            var applicationName = Assembly.GetExecutingAssembly().GetName().Name?.Replace(' ', '_');
```
That way, `applicationName` will be `null` which isn't that nice, but at least better than getting an exception in my opinion.

Reply:
Thanks I have made this change and also provided a default value of "NotAvailable" in case the result is null.